cmd/boulder-wfe2/main.go

```diff
@@ -555,20 +552,18 @@ func main() {
 		}()
 	}
 
-	done := make(chan bool)
-	go cmd.CatchSignals(logger, func() {
+	// When main is ready to exit (because it has received a shutdown signal),
+	// gracefully shutdown the servers. Calling these shutdown functions causes
+	// ListenAndServe() and ListenAndServeTLS() to immediately return, then waits
+	// for any lingering connection-handling goroutines to finish their work.
+	defer func() {
 		ctx, cancel := context.WithTimeout(context.Background(), c.WFE.ShutdownStopTimeout.Duration)
 		defer cancel()
-		_ = srv.Shutdown(ctx)
-		_ = tlsSrv.Shutdown(ctx)
-		done <- true
-	})
-
-	// https://godoc.org/net/http#Server.Shutdown:
-	// When Shutdown is called, Serve, ListenAndServe, and ListenAndServeTLS
-	// immediately return ErrServerClosed. Make sure the program doesn't exit and
-	// waits instead for Shutdown to return.
-	<-done
+		srv.Shutdown(ctx)
```

Comment:
I'm surprised you could remove the `_ =` assignments without triggering errcheck. These functions don't appear to be in our allowlist for errcheck. Do you know why this works? Could they have been added to some upstream default allowlist?

Reply:
Hmm, that's a really good point. Errcheck does have its own internal [list of default-excluded functions](https://github.com/kisielk/errcheck/blob/8f86be0cd56a95f4b0ddcbab3eec22f04e58b202/errcheck/excludes.go#L11-L52), but they're all from the standard library and don't include these methods.

But also, our errcheck excludes are done via regex, which is deprecated. Maybe something broke there, and we should switch to using errcheck's `exclude-functions:` mechanism instead. I'll put together a PR for that and see if it sheds any light on this.

Reply:
Oh! errcheck isn't in our list of enabled linters! I'll have a PR which re-enables it and fixes all of the places where errcheck bugs have snuck in tomorrow.

In the mean time, I've added the explicit underscore assignments here.

Reply:
https://github.com/letsencrypt/boulder/pull/6819